crates/biome_module_graph/src/format_module_graph.rs

```diff
@@ -0,0 +1,1096 @@
+        LineEnding::default()
+    }
+
+    fn as_print_options(&self) -> PrinterOptions {
+        PrinterOptions {
+            indent_width: self.indent_width(),
+            print_width: self.line_width().into(),
+            line_ending: self.line_ending(),
+            indent_style: self.indent_style(),
+        }
+    }
+}
+
+impl biome_formatter::FormatContext for ModuleFormatContext {
+    type Options = ModuleFormatContextOptions;
+
+    fn options(&self) -> &Self::Options {
+        &ModuleFormatContextOptions
+    }
+
+    fn source_map(&self) -> Option<&TransformSourceMap> {
+        None
+    }
+}
+
+impl std::fmt::Display for JsModuleInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let formatted = biome_formatter::format!(ModuleFormatContext, [&self.exports])
+            .expect("Formatting not to throw any FormatErrors");
+        f.write_str(
+            formatted
+                .print()
+                .expect("Expected a valid document")
+                .as_code(),
+        )
+    }
+}
+
+impl std::fmt::Display for Exports {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let formatted = biome_formatter::format!(ModuleFormatContext, [&self])
+            .expect("Formatting not to throw any FormatErrors");
+        f.write_str(
+            formatted
+                .print()
+                .expect("Expected a valid document")
+                .as_code(),
+        )
+    }
+}
+
+impl Format<ModuleFormatContext> for Exports {
+    fn fmt(
+        &self,
+        f: &mut biome_formatter::formatter::Formatter<ModuleFormatContext>,
+    ) -> FormatResult<()> {
+        let mut joiner = f.join();
+        for (export_name, export) in self.deref() {
+            let name = format_with(|f| match export_name {
+                Text::Borrowed(t) => {
+                    write!(
+                        f,
+                        [&format_args![
+                            text("Borrowed("),
+                            dynamic_text(&std::format!("{:?}", t.text()), TextSize::default()),
+                            text(")")
+                        ]]
+                    )
+                }
+                Text::Owned(t) => {
+                    write!(
+                        f,
+                        [&format_args![
+                            text("Owned("),
+                            dynamic_text(&std::format!("{:?}", t.as_str()), TextSize::default()),
+                            text(")")
+                        ]]
+                    )
+                }
+                Text::Static(t) => {
+                    write!(
+                        f,
+                        [&format_args![
+                            text("Static("),
+                            dynamic_text(&std::format!("{:?}", t), TextSize::default()),
+                            text(")")
+                        ]]
+                    )
+                }
+            });
```

Comment:
I think we can simplify this bit. It's probably not at all important which `Text` variant is being used, so we can just print the text itself only.